src/zenml/execution/pipeline/dynamic/runner.py

```diff
@@ -256,13 +278,24 @@ def launch_step(
             The step run outputs or a future for the step run outputs.
         """
         step = step.copy()
+        step_config = None
+        if self._run and self._run.triggered_by_deployment:
+            # Deployment-specific step overrides
+            step_config = StepConfigurationUpdate(
```

Comment:
One thing I forgot: We should also set `runtime="inline"` here for deployments, otherwise the steps might still launch in a remote environment using the orchestrator.